### DISCUSSION TITLE
How to add additional data to events?

### DISCUSSION THREAD

**Discussion by HellYeahOmg** [nextauthjs/next-auth#11427] [Help]
I need to pass down some client data stored in local storage and then use it on signIn events and such, but as far as i get, events are executed on server side? So Is it possible? I don't see any custom fields for this in the docs.

**Comment by k-taro56** [comment]
> events are executed on server side?
Yes, they are.

You can use Lazy Initialization to get the cookie attached to the request anywhere, not just at the sign-in event. localstorage information would be a good idea to add to the cookie.

https://authjs.dev/reference/nextjs#lazy-initialization
